bin/setup

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env sh
+
+fancy_echo() {
+  local fmt="$1"; shift
+
+  # shellcheck disable=SC2059
+  printf "\\n$fmt\\n" "$@"
+}
+
+install_linux_deps() {
+  local LINUX_DEPS="automake autoconf autogen libtool gcc curl cmake gnupg"
+  local flavor=`grep "^ID=" /etc/os-release | cut -d"=" -f 2`
+  echo $flavor
+  case $flavor in
+    debian|ubuntu)
+      sudo apt-get install -y $LINUX_DEPS libgmp-dev erlang-dev build-essential libssl-dev libncurses5-dev unzip lsb-release software-properties-common
+      ;;
+    fedora|centos)
+      sudo yum install $LINUX_DEPS gmp-devel
+      ;;
+    *)
+      fancy_echo "Unrecognized distribution $flavor"
+      exit 1
+      ;;
+  esac
+}
+
+install_linux_solc(){
+  echo "Installing Solc v0.4.25"
+  wget https://github.com/ethereum/solidity/releases/download/v0.4.25/solc-static-linux \
+  && chmod +x solc-static-linux \
+  && mv solc-static-linux solc  \
+  && sudo install -m0755 solc /usr/local/bin/solc
+}
+
+install_linux_postgres(){
+  echo "Installing Postgres 9.6."
+  echo "Adding public key 7FCC7D46ACCC4CF8 to repo."
+  sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 7FCC7D46ACCC4CF8
+  sudo add-apt-repository "deb http://apt.postgresql.org/pub/repos/apt/ $(lsb_release -sc)-pgdg main"
+  wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo apt-key add -
+  sudo apt-get update
+  sudo apt-get install postgresql-9.6 -y
+  sudo apt-get install postgresql-client-9.6
+  echo "Start PG."
+  sudo service postgresql start
+  echo "Create user."
+  sudo -i -u postgres psql -c "CREATE USER omisego_dev WITH PASSWORD 'omisego_dev';"
+  echo "Grant permission."
+  sudo -i -u postgres psql -c "ALTER USER omisego_dev CREATEDB;"
+  echo "Done with PG."
+}
+
+install_linux_geth(){
+  echo "Installing Geth 1.8.27"
+  mkdir gethDownload \
+  && wget -qO- https://gethstore.blob.core.windows.net/builds/geth-linux-amd64-1.8.27-4bcc0a37.tar.gz | tar xvz -C gethDownload --strip-components 1 \
+  && cd gethDownload \
+  && mkdir -p /usr/local/bin \
+  && sudo mv geth /usr/local/bin/ \
+  && cd .. \
+  && rm -r gethDownload
+}
+
+install_darwin_deps(){
+    fancy_echo "Installing dependencies via brew"
+    brew bundle --file=- <<EOF
+brew "automake"
+brew "autoconf"
+brew "gmp"
+brew "libtool"
+brew "cmake"
+EOF
+}
+
+install_deps() {
+  local sys=`uname -s`
+  echo $sys
+  case $sys in
+    Linux*)
+      install_linux_deps
+      install_linux_solc
+      install_linux_geth
+      if [ -z "$CIRLCECI" ]; then
+        echo "Skip Postgres setup, because CircleCI."
+      else
+        install_linux_postgres
+      fi
+      ;;
+    Darwin*)
+      install_darwin_deps
```

Comment:
No geth, solc, postgres setup for mac?

Reply:
yeah... the problematic part is I have no idea how to test it. so rather then including something I'm not 100% sure it works, I left it open...